### PR TITLE
Sprint4/integration refactor

### PR DESCRIPTION
**Pull request by Mattioo7**

### CODE REVIEW

File: app/Filters.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ export default function ModalScreen() {
 				onFocus={() => setIsFocusCategory(true)}
 				onBlur={() => setIsFocusCategory(false)}
 				onChange={item => {
-					setCategoryId(item.id);
+					setCategoryId(prev => item.id);
```
Comment (jacob-b-labs) [inline app/Filters.tsx:85]:
```suggestion
					setCategoryId(item.id);
```
We use lambda only when we want to transform an existing state. Old form was better.

File: app/event.tsx [hunk 2]
```diff
@@ -69,19 +72,20 @@ export default function ModalScreen() {
...
 
 				// refetch event
-				setLoading(true);
 				getEvent()
 					.then(() => {
 						setLoading(false);
+					})
+					.then(() => {
+						Alert.alert('Reservation successful');
 					});
```
Comment (jacob-b-labs) [inline app/event.tsx:87]:
```suggestion
				getEvent()
					.then(() => {
						Alert.alert('Reservation successful');
					})
					.finally(() => {
						setLoading(false);
					});
```
If getEvent() fails, we don't want `loading` to stay forever.

**Review by jacob-b-labs — CHANGES_REQUESTED** [review]
All good, minor fixes are needed

File: app/Filters.tsx (hunk 1 above)
Comment (Mattioo7) [reply app/Filters.tsx]: Done

File: app/event.tsx (hunk 2 above)
Comment (Mattioo7) [reply app/event.tsx]: Done

**Review by jacob-b-labs — APPROVED** [review]

**Pull request merged**